invoice/src/main/java/org/killbill/billing/invoice/usage/SubscriptionUsageInArrear.java

```diff
@@ -40,7 +40,7 @@
 import org.killbill.billing.invoice.generator.InvoiceWithMetadata.TrackingRecordId;
 import org.killbill.billing.invoice.usage.ContiguousIntervalUsageInArrear.UsageInArrearItemsAndNextNotificationDate;
 import org.killbill.billing.junction.BillingEvent;
-import org.killbill.billing.usage.RawUsage;
+import org.killbill.billing.usage.plugin.api.RawUsageRecord;
```

Comment:
Do we want the internal modules to rely on `plugin-api` objects explicitly? Shouldn't the `usage` module transforms `org.killbill.billing.usage.plugin.api.RawUsageRecord` into a non-plugin version?

For the catalog for instance, only the `catalog` module relies on `VersionedPluginCatalog` (from `plugin-api`) and transforms it into a `DefaultVersionedCatalog`.

Reply:
Yes, i went back and forth with that. Fully agree it would be cleaner, and was expecting to discuss that with you.

The benefits of using the RawUsageRecord is to avoid copying/converting every single usage record into a new model that is likely to be the exact same one. In the case of the catalog, we had no choice because the catalog objects are not dummy POJO, they have a lot of business logics encapsulated. In the usage case, those are strictly POJO. I think a better approach would be to define RawUsageRecord in our api module and use it both in the plugin api and KB core, but before i embark in this change i wanted to have the conversation.

Reply:
> In the usage case, those are strictly POJO. I think a better approach would be to define RawUsageRecord in our api module and use it both in the plugin api and KB core, but before i embark in this change i wanted to have the conversation.

Let's talk, but I think it's a cleaner approach indeed.